src/common/nlmeans_core.c

```diff
@@ -510,7 +510,7 @@ void nlmeans_denoise(const float *const inbuf, float *const outbuf,
           }
           const int pcol_min = chunk_left - MIN(radius,MIN(chunk_left,chunk_left+scol));
           const int pcol_max = chunk_right + MIN(radius,MIN(width-chunk_right,width-(chunk_right+scol)));
-          if (row < row_top)
+          if (row < MIN(row_top, row_bot))
```

Comment:
I don't understand the semantic here. Don't you meant:

`if(row < row_top || row > row_bot)`

Reply:
I think it's correct that the check is for `MIN(row_top, row_bot)` because it might be possible that bottom bound lies above top bound and we do not want to exceed either...

Reply:
Maybe, but I'll wait for confirmation from @ralfbrown to be sure.

Reply:
Johnny-bit is correct, the cause of the crash was that row_top was greater than row_bot.

Reply:
Probably an artifact of the algorithm as in all the code I went through the top was always smaller than the bottom as origin is on top-left. Anyway, merging then. Thanks!

Reply:
In the particular instance I managed to catch a segfault, the patch was at a vertical offset of 896 on a 900px high image, with a patch radius of 2.  After constraining things to stay within the input image, that resulted in row_top of 898 and row_bot of 897....